### PR TITLE
refactor(DatePicker,DateRangePicker): rename prop disabledDate -> isDateDisabled

### PR DESCRIPTION
**Pull request by SevenOutman**
Reduce ambiguousness

### CODE REVIEW

**Comment by simonguo** [comment]
If you want to change the name of `disabledDate`, you need to consider the names of the names of `disabledHours`, `disabledMinutes`, and `disabledSeconds` also changed, so as to maintain consistency.

In addition, I recommend using the method name of the method of `shouldDisableDate`, refer to the [MUI API](https://mui.com/x/api/date-pickers/date-picker/#props).

**Comment by SevenOutman** [comment]
> If you want to change the name of `disabledDate`, you need to consider the names of the names of `disabledHours`, `disabledMinutes`, and `disabledSeconds` also changed, so as to maintain consistency.

OK, I'll rename them too.

> In addition, I recommend using the method name of the method of `shouldDisableDate`, refer to the [MUI API](https://mui.com/x/api/date-pickers/date-picker/#props).

The "should" makes more sense to me. I'll use that.

**Review by simonguo — APPROVED** [review]
The documentation needs to be updated after the version is released.

File: src/utils/dateUtils.ts [hunk 1]
```diff
@@ -66,20 +66,21 @@ function validTime(calendarProps: any, date: Date) {
 
   return Object.keys(calendarProps).some(key => {
     if (/(Hours)/.test(key)) {
-      return calendarProps[key](getHours(date), date);
+      return calendarProps[key]?.(getHours(date), date) ?? true;
```
Comment (simonguo) [inline src/utils/dateUtils.ts:69]: The default is `true`, which will cause the time option to always be in a warning state.
image

**Pull request merged**